src/contrib/cluster/Akka.Cluster.Sharding.Tests.MultiNode/ClusterShardingMinMembersSpec.cs

```diff
@@ -0,0 +1,243 @@
+            }, _config.First, _config.Second, _config.Third);
+            EnterBarrier("after-1");
+
+            RunOn(() =>
+            {
+                //check persistence running
+                var probe = CreateTestProbe();
+                var journal = Persistence.Persistence.Instance.Get(Sys).JournalFor(null);
+                journal.Tell(new Persistence.ReplayMessages(0, 0, long.MaxValue, Guid.NewGuid().ToString(), probe.Ref));
+                probe.ExpectMsg<Persistence.RecoverySuccess>(TimeSpan.FromSeconds(10));
+            }, _config.First, _config.Second, _config.Third);
+            EnterBarrier("after-1-test");
+        }
+
+        public void Cluster_with_min_nr_of_members_using_sharding_should_use_all_nodes()
+        {
+            Within(TimeSpan.FromSeconds(30), () =>
+            {
+                Join(_config.First, _config.First);
+                RunOn(() =>
+                {
+                    StartSharding();
+                }, _config.First);
+
+                Join(_config.Second, _config.First);
+                RunOn(() =>
+                {
+                    StartSharding();
+                }, _config.Second);
+
+                Join(_config.Third, _config.First);
+                // wait with starting sharding on third
+                Within(Remaining, () =>
+                {
+                    AwaitAssert(() =>
+                    {
+                        Cluster.State.Members.Count.Should().Be(3);
+                        Cluster.State.Members.Should().OnlyContain(i => i.Status == MemberStatus.Up);
+                    });
+                });
+                EnterBarrier("all-up");
+
+                RunOn(() =>
+                {
+                    _region.Value.Tell(1);
+                    // not allocated because third has not registered yet
+                    ExpectNoMsg(TimeSpan.FromSeconds(2));
+                }, _config.First);
+                EnterBarrier("verified");
+
+                RunOn(() =>
+                {
+                    StartSharding();
+                }, _config.Third);
+
+                RunOn(() =>
+                {
+                    // the 1 was sent above
+                    ExpectMsg(1);
+                    _region.Value.Tell(2);
+                    ExpectMsg(2);
+                    _region.Value.Tell(3);
+                    ExpectMsg(3);
+                }, _config.First);
+                EnterBarrier("shards-allocated");
+
+                _region.Value.Tell(new GetClusterShardingStats(Remaining));
+
+                var stats = ExpectMsg<ClusterShardingStats>();
+                var firstAddress = Node(_config.First).Address;
+                var secondAddress = Node(_config.Second).Address;
+                var thirdAddress = Node(_config.Third).Address;
+
+                stats.Regions.Keys.Should().BeEquivalentTo(new Address[] { firstAddress, secondAddress, thirdAddress });
```

Comment:
Still not sure how yo use it in this context, since hint is only supported for ExpectMsg or similar. 
Here i can only provide "because" parameter, is it the same?